lib/src/protocol/http/parser/mod.rs

```diff
@@ -263,11 +263,12 @@ fn status_line(i: &[u8]) -> IResult<&[u8], StatusLine> {
 }
 
 #[derive(PartialEq, Debug)]
-pub struct Header<'a> {
-    pub name: &'a [u8],
-    pub value: &'a [u8],
+pub struct Header {
+    pub name: Vec<u8>,
+    pub value: Vec<u8>,
 }
 
+#[cfg(feature = "tolerant-http1-parser")]
```

Comment:
the fix for this should probably be available even under the `tolerant-http1-parser` feature. That feature is only there to accept IS0-8859-1 characters in header values, something that should not exist anymore (but for prod reasons we had to add it)